vowpalwabbit/get_pmf.cc

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#include "get_pmf.h"
+#include "err_constants.h"
+#include "api_status.h"
+#include "debug_log.h"
+#include "parse_args.h"
+
+// Aliases
+using VW::LEARNER::single_learner;
+using std::endl;
+using VW::cb_continuous::continuous_label;
+using VW::cb_continuous::continuous_label_elm;
+using VW::config::make_option;
+using VW::config::option_group_definition;
+using VW::config::options_i;
+
+// Enable/Disable indented debug statements
+VW_DEBUG_ENABLE(false)
+
+namespace VW
+{
+namespace continuous_action
+{
+////////////////////////////////////////////////////
+// BEGIN sample_pdf reduction and reduction methods
+struct get_pmf
+{
+  int learn(example& ec, api_status* status);
+  int predict(example& ec, api_status* status);
+
+  void init(single_learner* p_base, float epsilon);
+
+ private:
+  single_learner* _base = nullptr;
+  float _epsilon;
+};
+
+int get_pmf::learn(example& ec, api_status*)
+{
+  _base->learn(ec);
+  return error_code::success;
+}
+
+int get_pmf::predict(example& ec, api_status*)
+{
+  uint32_t base_prediction;
+
+  {  // predict & restore prediction
+    restore_prediction restore(ec);
```

Comment:
I believe this functionality is already provided by the `swap_guard` can you please look at using that instaed?

Reply:
Yes.  Will check it out.

Reply:
Changed to swap_guard